app/src/main/java/com/github/epfl/meili/util/MenuInflaterHelper.kt

```diff
@@ -0,0 +1,61 @@
+package com.github.epfl.meili.util
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.view.Menu
+import android.view.MenuItem
+import androidx.core.content.ContextCompat.startActivity
+import com.github.epfl.meili.R
+import com.github.epfl.meili.forum.ForumActivity
+import com.github.epfl.meili.map.MapActivity
+import com.github.epfl.meili.messages.ChatLogActivity
+import com.github.epfl.meili.models.PointOfInterest
+import com.github.epfl.meili.review.ReviewsActivity
+
+object MenuInflaterHelper {
+    fun onCreateOptionsMenuHelper(activity: Activity, ctx: Int, menu: Menu?){
+        activity.menuInflater.inflate(ctx, menu)
+    }
+
+    fun onOptionsItemSelectedHelpoer(acitvity: Activity,item: MenuItem, intent: Intent){
+        // get the POI
+        val poi = intent.getParcelableExtra<PointOfInterest>(MapActivity.POI_KEY)!!
+
+        //Now that the buttons are added at the top control what each menu buttons does
+        val initializedIntent: Intent = when (item?.itemId) {
+            R.id.menu_reviews_from_forum -> {
+                Intent(acitvity, ReviewsActivity::class.java)
+                    .putExtra(MapActivity.POI_KEY, poi)
+            }
+            R.id.menu_chat_from_forum-> {
+                Intent(acitvity, ChatLogActivity::class.java)
+                    .putExtra(MapActivity.POI_KEY, poi)
+            }
+            R.id.menu_reviews_from_chat -> {
+                Intent(acitvity, ReviewsActivity::class.java)
+                    .putExtra(MapActivity.POI_KEY, poi)
+            }
+            R.id.menu_forum_from_chat-> {
+                Intent(acitvity, ForumActivity::class.java)
+                    .putExtra(MapActivity.POI_KEY, poi)
+            }
+            R.id.menu_chat_from_review -> {
+                Intent(acitvity, ChatLogActivity::class.java)
+                    .putExtra(MapActivity.POI_KEY, poi)
+            }
+            R.id.menu_forum_from_review-> {
+                Intent(acitvity, ForumActivity::class.java)
+                    .putExtra(MapActivity.POI_KEY, poi)
+            }
+            else -> {
+                Intent(acitvity, ForumActivity::class.java)
+            }
+        }
```

Comment:
I think we could replace menu_chat_from_review and menu_chat_from_forum with buttons with the same id menu_chat. I checked and we can indeed have buttons with the same id as long as they are in different layouts. This will remove the duplication in this part of the code. What do you think ?

Reply:
Sorry, my comment was on outdated code. What you did already looks better! But I think we could still do this, for the sake of having the same id everywhere.

Reply:
Also, I think we should not instantiate the activity but rather store launchedActivity::class.java directly. For eg. ReviewsActivity::class.java

Reply:
Both done 👍 I wasn't sure about the buttons having the same name but you're right it works